src/js/app.js

```diff
@@ -1,17 +1,153 @@
+    upload_btn.addEventListener("click", async function(e) {  
+      if (App.upload_image == null) {
+        alert("no file chosen");
+        return;
+      }
+      console.log(App.upload_image);
+      // request S3 upload URI
+      var presignedURL;
+      await fetch(App.API_ENDPOINT, {
+        method: 'GET'
+      }).then(response => {
+        resp_body = response.text()
+        resp_body.then((res)=>{
+          presignedURL = res
+        }).then(function() {
+          console.log(presignedURL)
+          let binary = atob(App.upload_image.split(',')[1])
+          let array = []
+          for (var i = 0; i < binary.length; i++) {
+            array.push(binary.charCodeAt(i))
+          }
+          blobData = new Blob([new Uint8Array(array)], {type: 'image/jpeg'})
+        }).then(async function() {
+          // upload to s3
+          presignedURL = JSON.parse(presignedURL)
+          await fetch(presignedURL['uploadURL'], {
+            method: 'PUT',
+            body: blobData
+          }).then(res => console.log("upload result", res)).then(async function() {
+            var resp_body
+            var textract_res
+            
+            
+            var image_url = presignedURL['uploadURL'].split('?')[0];
+
+            var pic_info =  {
+              "uuid": uuidv4(),
+              "pic-url": image_url,
+              "uploader": "Jara",
+              "uploader-id": 0
+            }
+
+            await fetch(App.TEXTRACT_ENDPOINT, {
+              method: 'POST',
+              body: JSON.stringify(pic_info)
+            }).then((response) => {
+              resp_body = response.json()
+              resp_body.then((res)=>{
+                // console.log(res)
+                textract_res = res
+              }).then(async function() {
+                pic_info.bibs = textract_res
+                // remove duplicates in textract result
+                console.log(textract_res)
+                textract_res = [...new Set(textract_res)]
+
+                console.log(textract_res)
+  
+                let id = pic_info.id
+                for (let i = 0; i < textract_res.length - 1; i++) {
+                  // use fixed id=1 since it doesn't affect the result
+                  let integer = parseInt(textract_res[i])
+                  var uuid_each = uuidv4()
+                  console.log(uuid_each)
+                  let data_2_insert = {
+                      "id": 888,
+                      "bib": integer,   
+                      "url": pic_info['pic-url'],
+                      "photographer": "JaRa"
+                    }
+                  
+                  entry = {}
+                  entry[uuid_each] = data_2_insert
+                  App.updateStore(entry)
+                  // send subscription email
+                  // post to azure pub/sub service
+                  await fetch(App.EMAIL_GENERATE_EVENT, {
+                    method: 'POST',
+                    body: {
```

Comment:
`// post to azure pub/sub servic`
 `                 await fetch(App.EMAIL_GENERATE_EVENT, {`
 `                   method: 'POST',`
 `                   body: {`
stringify the body object

Reply:
just stringified